engine/order_manager.go

```diff
@@ -108,21 +109,16 @@ func (m *OrderManager) gracefulShutdown() {
 // run will periodically process orders
 func (m *OrderManager) run() {
 	log.Debugln(log.OrderMgr, "Order manager started.")
-	timer := time.NewTimer(orderManagerDelay)
 	for {
 		select {
 		case <-m.shutdown:
 			m.gracefulShutdown()
-			if !timer.Stop() {
-				<-timer.C
-			}
 			m.orderStore.wg.Done()
 			log.Debugln(log.OrderMgr, "Order manager shutdown.")
 			return
-		case <-timer.C:
+		case <-time.After(orderManagerInterval):
```

Comment:
Happy to separate this out from the fix, because it's not atomic, it needs discussion, and the PR won't be atomic this 
 way.
 
The channel not draining during shutdown won't be an issue because it won't prevent or delay shutdown.
It just won't be GC'd, but that's a minor leak during a shutdown (IMO)

The reasoning for the issue I'm pointing to is that Reset *isn't* called directly.
The race would be that 10s occurs between the go m.processOrders and the Reset, due to context switching and then starvation or suspension.

I'm really not fussed though. If it's not a slam dunk let's drop it, it's a super edge case and very unlikely.
And the result wouldn't even be that bad.

Yanking it from this PR. Shout if I've swayed you and I should submit it in a separate PR 👍